wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java

```diff
@@ -251,6 +254,18 @@ public MachineProcessDto executeCommandInMachine(@ApiParam(value = "Machine ID")
                              .collect(Collectors.toList());
     }
 
+    @GET
+    @Path("/{machineId}/recipe")
+    @Produces(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Retrieve machine recipe script")
+    @ApiResponses({@ApiResponse(code = 200, message = "Recipe content retrieved"),
+                   @ApiResponse(code = 404, message = "Machine with specified ID does not exist"),
+                   @ApiResponse(code = 500, message = "Internal server error occurred")})
+    public String getRecipeScript(@ApiParam(value = "Machine ID") @PathParam("machineId") String id)
+            throws NotFoundException, ServerException {
+        return recipeRetriever.getRecipe(machineManager.getMachine(id).getConfig()).getScript();
```

Comment:
-1 to add that to machine service. This functionality is needed for client only and tightly coupled with client-side limitations. Also it makes refactoring/removal of machine service harder. It is also can conflict with upcoming changes of environments. 
For such functionality I suggest creation of reset service not in API area but in IDE specific area.